sympy/solvers/tests/test_solvers.py

```diff
@@ -644,8 +644,11 @@ def test_solve_transcendental():
         assert (sol[0][y] + sol[1][y]).is_Rational, (yi,sol)
     # don't allow massive expansion
     assert solve(cos(1000*x) - S.Half) == [pi/3000, pi/600]
-    assert solve(cos(x - 1000*y) - 1, x) == [2*atan(tan(500*y))]
-    assert solve(cos(x + y + z) - 1, x) == [-2*atan(tan(y/2 + z/2))]
+    assert solve(cos(x - 1000*y) - 1, x) == [1000*y, 1000*y + 2*pi]
+    assert solve(cos(x + y + z) - 1, x) == [-y - z, -y - z + 2*pi]
```

Comment:
This is an improvement and it is consistent with:
```
In [13]: solve(cos(x)-1, x)
Out[13]: [0, 2⋅π]
```
Why though does this return two solutions?

I would have expected that periodic solutions be returned in a half-open range like `[0, 2*pi)`.

Reply:
Note that `solvify` does this:
```python
In [22]: solvify(cos(x)-1, x, Reals)
Out[22]: [0]
```

Reply:
I don't recall a specific filtering of the result in `solve` but maybe `solvify` does the extra test to remove periodic solutions.

Reply:
It happens here:
https://github.com/sympy/sympy/blob/f45524749ae30837be99f5dadcb25626ec36cbda/sympy/solvers/solvers.py#L2630-L2634
https://github.com/sympy/sympy/blob/f45524749ae30837be99f5dadcb25626ec36cbda/sympy/solvers/solvers.py#L2795-L2801
This code does not know the value of rhs though:
```
(Pdb) p soln
[acos(_t), -acos(_t) + 2*pi]
```